### PR TITLE
TELECOM-9999: SO_MARK connections associated with specific listener

### PR DESCRIPTION
**Pull request by benceszigeti**

### CODE REVIEW

**Comment by ghmj2417** [comment]
Will the mark be configurable? Let's say I wanted to use 7 for transaction instances, but 2 for forward instances?

**Comment by benceszigeti** [comment]
> Will the mark be configurable? Let's say I wanted to use 7 for transaction instances, but 2 for forward instances?
Yes, we can do `socket = tls:1.2.3.4:1234 mark 0x2` for forward listener, and `socket = tls:1.2.3.4:1234 mark 0x7` for transaction listener, so can set them separately.

File: net/tcp_common.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ int tcp_connect_blocking_timeout(int fd, const struct sockaddr *servaddr,
 
 
 int tcp_sync_connect_fd(union sockaddr_union* src, union sockaddr_union* dst,
-                 enum sip_protos proto, struct tcp_conn_profile *prof, enum si_flags flags);
+                 enum sip_protos proto, struct tcp_conn_profile *prof, enum si_flags flags, int mark);
```
Comment (davidtrihy-genesys) [inline net/tcp_common.h:34]: the mark being passed in the function is not being used here, seems like it only needs to be used on setsockopt

**Review by davidtrihy-genesys — APPROVED** [review]

**Comment by davidtrihy-genesys** [comment]
Merged master into this branch after merging my PR and doing a build to test against the Envoy branch

**Pull request merged**